src/kademlia/dht_storage.cpp

```diff
@@ -247,6 +247,10 @@ namespace
 					// peer list
 					if (noseed && iter->seed) continue;
 
+					// only include peers with the right address family
+					if (iter->addr.protocol().family() != protocol.family())
```

Comment:
The `want` argument only applies to the `nodes`/`nodes6` keys. The `value` key always contains peers with the same address family as the request was received from.